### PR TITLE
Simplify adding groups with via the FixtureGroupInterface

### PR DESCRIPTION
**Pull request by TerjeBr**

### CODE REVIEW

**Comment by weaverryan** [comment]
Hey!

I saw your comment about this in my other PR. I understand your argument about tying to simplify the compiler pass. But you also mentioned that the original version of the code might be wron in some cases. Could you explain more?

File: Loader/SymfonyFixturesLoader.php [hunk 1]
```diff
@@ -28,14 +29,17 @@
...
-            $class = get_class($fixture['fixture']);
-            $this->loadedFixtures[$class] = $fixture['fixture'];
-            $this->addGroupsFixtureMapping($class, $fixture['groups']);
+        // Because parent::addFixture may call $this->createFixture
+        // we cannot call $this->addFixture in this loop
+        foreach ($fixtures as $fixtureData) {
+            $fixture = $fixtureData['fixture'];
+            $class = get_class($fixture);
+            $this->addGroupsFixtureMapping($class, $fixtureData['groups']);
+            $this->loadedFixtures[$class] = $fixture;
```
Comment (weaverryan) [inline Loader/SymfonyFixturesLoader.php]: Did you change anything here? Or just reorder and rename things? If you just renamed things, could we change it back? It’s hard to focus on the important things that you added. But if you did make a significant change here, could you explain it?

Thanks :)

**Comment by TerjeBr** [comment]
The original version could fail if a fixture service was defined by a factory method, and the actual class of the service/fixture was different from the declared class in the container. F.ex. it often happens that factory methods return different subclasses of the declared class. If the declared class does not implement FixtureGroupInterface but the child class does, then the original code will not find the groups in the child class.

File: Loader/SymfonyFixturesLoader.php (hunk 1 above)
Comment (TerjeBr) [reply Loader/SymfonyFixturesLoader.php]: Yes, you are right. It was over a couple of commits. (If you follow my commits you will see.) But yes I ended up just reordering and renaming that part. Sorry about that, was not my intention.

**Pull request merged**